### PR TITLE
routeOnLoad option

### PR DESCRIPTION
**Pull request by evanpipta**
Added an optional parameter to not fire a route event on page load. I couldn't get it to work properly with server side rendering without adding this.

### CODE REVIEW

File: src/es-router.js [hunk 1]
```diff
@@ -73,7 +73,9 @@ class EsRouter {
     }
 
     //do an initial routing
-    this.path(this.getPathFromUrl());
+    if (routeOnLoad !== false) {
```
Comment (joshbDev) [inline src/es-router.js]: you can just have `if (routeOnLoad) {`

**Comment by joshbDev** [comment]
just the one comment, then
```

                       _____|\
                  _.--| SSt |:
                 <____|.----||
                        .---''---,
                         ;..__..'    _...
                       ,'/  ;|/..--''    \
                     ,'_/.-/':            :
                _..-'''/  /  |  \    \   _|/|
               \      /-./_ \;   \    \,;'   \
               ,\    / \:  `:\    \   //    `:`.
             ,'  \  /-._;   | :    : ::    ,.   .
           ,'     ::   /`-._| |    | || ' :  `.`.)
        _,'       |;._:: |  | |    | `|   :    `'
      ,'   `.     /   |`-:_ ; |    |  |  : \
      `--.   )   /|-._:    :          |   \ \
         /  /   :_|   ;`-._;   __..--';    : :
        /  (    ;|;-./_  _/.-:'o |   /     ' |
       /  , \._/_/_./--''/_|:|___|_,'        |
      :  /   `'-'--'----'---------'          |
      | :     O ._O   O_. O ._O   O_.      ; ;
      : `.      //    //    //    //     ,' /
    ~~~`.______//____//____//____//_______,'~
              //    //~   //    //
       ~~   _//   _//   _// ~ _//     ~
     ~     / /   / /   / /   / /  ~      ~~
          ~~~   ~~~   ~~~   ~~~

```
Ship it!

File: README.md [hunk 2]
```diff
@@ -52,6 +53,9 @@ this gives you the ability to go to any path you'd like, regardless of whether y
 ### base (required unless useHash is false)
 this is the base url for your application if you aren't using the hash. if this isn't declared and you are using `useHash`, it will try and retrieve the base from the `base` tag in the html5Mode
 
+### routeOnLoad (optional, defauls to true)
+this makes a route change event get fired upon loading the page. if this is set to false, no initial route change will be fired. setting this to false may work better for some app configurations that use server side rendering.
```
Comment (chrisrng) [inline README.md]: nit: capitalize "Setting" for consistency

File: src/es-router.js (hunk 1 above)
Comment (evanpipta) [reply src/es-router.js]: Doesn't that make it false by default then?

Comment (joshbDev) [reply src/es-router.js]: ah, yeah, though then on `routeOnLoad` you should do`, routeOnLoad = true}`, that would make more logical sense

**Comment by evanpipta** [comment]
Updated the pull request

**Comment by joshbDev** [comment]
ship it!

**Review by chrisrng — APPROVED** [review]
lgtm

**Comment by evanpipta** [comment]
I don't have admin access on the repo

**Comment by joshbDev** [comment]
@747823 unfortunately I've been removed from permissions on the repo as well, you'll have to find someone in the Priceline org to merge

**Comment by chrisrng** [comment]
I gave @Emuentes admin before I left :)

**Comment by Emuentes** [comment]
Good to go @747823

**Pull request merged**